README.md

```diff
@@ -2,92 +2,707 @@
+/ospool/uc-shared/public/data.txt: {NamespacePath: https://cache1.example.com/ospool/uc-shared/public/data.txt, Success: True} {NamespacePath: https://cache1.example.com/ospool/uc-shared/public/data.txt, Success: True} {NamespacePath: https://cache2.example.com/ospool/uc-shared/public/data.txt, Success: False, Error: <class 'aiohttp.client_exceptions.ClientConnectorError'>}
+```
+
+**What the statistics show:**
+- **NamespacePath**: The full cache URL that was accessed
+- **Success**: Whether the cache access succeeded (`True`) or failed (`False`)
+- **Error**: The exception type if the access failed (only shown on failures)
+
+This helps identify:
+- Which caches are being used for your requests
+- Cache reliability and failure patterns
+- Whether cache fallback is working correctly
+
+### Enabling Debug Logging
+
+Enable detailed logging to troubleshoot issues:
+
+```python
+import logging
+
+# Set logging level for PelicanFS
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger("fsspec.pelican")
+logger.setLevel(logging.DEBUG)
+```
+
+## API Reference
+
+### PelicanFileSystem
+
+Main class for interacting with Pelican federations.
+
+#### Constructor Parameters
+
+- `federation_discovery_url` (str): The Pelican federation discovery URL (e.g., `"pelican://osg-htc.org"`)
+- `direct_reads` (bool, optional): If `True`, read directly from origins instead of caches. Default: `False`
+- `preferred_caches` (list, optional): List of preferred cache URLs. Use `"+"` to append director's caches
+- `headers` (dict, optional): HTTP headers to include in requests. Use for authorization: `{"Authorization": "Bearer TOKEN"}`
+- `use_listings_cache` (bool, optional): Enable caching of directory listings. Default: `False`
+- `asynchronous` (bool, optional): Use async mode. Default: `False`
+- `**kwargs`: Additional arguments passed to the underlying HTTP filesystem
+
+#### Methods
+
+##### Object Operations
+
+- `ls(path, detail=True, **kwargs)` - List objects in a collection
+- `cat(path, recursive=False, on_error="raise", **kwargs)` - Read object contents
+- `open(path, mode, **kwargs)` - Open an object for reading or writing
+- `glob(path, maxdepth=None, **kwargs)` - Find objects matching a pattern
+- `find(path, maxdepth=None, withdirs=False, **kwargs)` - Recursively list all objects
+- `put(lpath, rpath, recursive=False, **kwargs)` - Upload local file(s) as remote object(s)
+- `get(rpath, lpath, recursive=False, **kwargs)` - Download remote object(s) to local file(s)
+
+##### Utility Methods
+
+- `get_access_data()` - Get cache access statistics
+- `info(path, **kwargs)` - Get detailed information about an object
+- `exists(path, **kwargs)` - Check if a path exists
+- `isfile(path, **kwargs)` - Check if a path is an object
+- `isdir(path, **kwargs)` - Check if a path is a collection
+
+### OSDFFileSystem
+
+Convenience class that automatically connects to the OSDF federation (`osg-htc.org`).
```

Comment:
I'd change "(`osg-htc.org`)" to "(which uses `osg-htc.org` for its discovery URL)". This re-enforces the concept of a discovery URL for the reader.